File: kotlin-camunda-webcast-app/src/main/java/de/viadee/camunda/demo/dto/AddressDTO.java
package de.viadee.camunda.demo.dto;

import java.util.Objects;

public class AddressDTO {
    private String street;
    private String zipCode;
    private String city;
    private String country;

    public String getStreet() {
        return street;
    }

    public void setStreet(final String street) {
        this.street = street;
    }

    public String getZipCode() {
        return zipCode;
    }

    public void setZipCode(final String zipCode) {
        this.zipCode = zipCode;
    }

    public String getCity() {
        return city;
    }

    public void setCity(final String city) {
        this.city = city;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(final String country) {
        this.country = country;
    }

    @Override
    public boolean equals(final Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        final AddressDTO address = (AddressDTO) o;
        return Objects.equals(street, address.street) && Objects.equals(zipCode, address.zipCode) && Objects.equals(city, address.city) && Objects.equals(country, address.country);
    }

    @Override
    public int hashCode() {
        return Objects.hash(street, zipCode, city, country);
    }

    @Override
    public String toString() {
        return "Address{" +
                "street='" + street + '\'' +
                ", zipCode='" + zipCode + '\'' +
                ", city='" + city + '\'' +
                ", country='" + country + '\'' +
                '}';
    }
}

File: kotlin-camunda-webcast-kotlin/src/main/kotlin/de/viadee/camunda/demo/dto/AddressDTO.kt
package de.viadee.camunda.demo.dto

import java.util.*

class AddressDTO {
    var street: String? = null
    var zipCode: String? = null
    var city: String? = null
    var country: String? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val address = other as AddressDTO
        return street == address.street && zipCode == address.zipCode && city == address.city && country == address.country
    }

    override fun hashCode(): Int {
        return Objects.hash(street, zipCode, city, country)
    }

    override fun toString(): String {
        return "Address{" +
                "street='" + street + '\'' +
                ", zipCode='" + zipCode + '\'' +
                ", city='" + city + '\'' +
                ", country='" + country + '\'' +
                '}'
    }
}